matter_server/server/helpers/paa_certificates.py

```diff
@@ -22,10 +23,57 @@
 PRODUCTION_URL = "https://on.dcl.csa-iot.org"
 TEST_URL = "https://on.test-net.dcl.csa-iot.org"
 GIT_URL = "https://github.com/project-chip/connectedhomeip/raw/master/credentials/development/paa-root-certs"  # pylint: disable=line-too-long
-GIT_CERTS = [
-    "Chip-Test-PAA-FFF1-Cert",
-    "Chip-Test-PAA-NoVID-Cert",
-]
+
+# Git repo details
+OWNER = "project-chip"
+REPO = "connectedhomeip"
+PATH = "credentials/development/paa-root-certs"
+
+
+async def get_directory_contents(owner: str, repo: str, path: str) -> List[str]:
+    """
+    Fetch directory contents from a GitHub repository.
+
+    Args:
+        owner (str): The owner of the GitHub repository.
+        repo (str): The name of the GitHub repository.
+        path (str): The path within the repository.
+
+    Returns:
+        list: A list of file names in the specified directory.
+    """
+    api_url = f"https://api.github.com/repos/{owner}/{repo}/contents/{path}"
+    async with ClientSession() as session:
+        async with session.get(api_url, timeout=20) as response:
+            if response.status == 200:
+                contents = await response.json()
+                return [item["name"] for item in contents]
+
+            LOGGER.error(
+                "Failed to fetch directory contents. Status code: %s", response.status
+            )
+            return []
+
+
+async def get_git_file_list() -> List[str]:
+    """
+    Retrieve a list of unique file names from a Git repository.
+
+    This function fetches the list of file names from a specified path in a Git repository,
+    filters out the file extensions, and returns a list of unique file names.
+
+    Returns:
+        List[str]: A list of unique file names.
+    """
+    file_list = await get_directory_contents(OWNER, REPO, PATH)
+    # Filter out extension and remove duplicates
+    unique_file_names = list({file.split(".")[0] for file in file_list})
+    return unique_file_names
+
+
+GIT_CERTS = asyncio.run(get_git_file_list())
```

Comment:
I'd prefer fetching these as part of `fetch_git_certificates` so we don't have side effects on import of this module.

Reply:
Ack. Made changes - testing it. Will deploy to patch after testing in my HA